modules/src/ics04_channel/handler/chan_open_confirm.rs

```diff
@@ -131,73 +127,43 @@ mod tests {
             msg: ChannelMsg,
             want_pass: bool,
         }
-        let proof_height = 10;
-        let client_consensus_state_height = 10;
-        let host_chain_height = Height::new(1, 35);
-
+        let client_id = ClientId::new(ClientType::Mock, 24).unwrap();
+        let conn_id = ConnectionId::new(2).unwrap();
         let context = MockContext::default();
+        let client_consensus_state_height = context.host_current_height().revision_height;
 
-        let msg_conn_try = MsgConnectionOpenTry::try_from(get_dummy_msg_conn_open_try(
-            client_consensus_state_height,
-            host_chain_height.revision_height,
-        ))
-        .unwrap();
-
+        // The connection underlying the channel we're trying to open.
         let conn_end = ConnectionEnd::new(
             ConnectionState::Open,
-            msg_conn_try.client_id().clone(),
-            ConnectionCounterparty::new(
-                msg_conn_try.counterparty().client_id().clone(),
-                Some(ConnectionId::from_str("defaultConnection-0").unwrap()),
-                msg_conn_try.counterparty().prefix().clone(),
-            ),
+            client_id.clone(),
+            ConnectionCounterparty::try_from(get_dummy_raw_counterparty()).unwrap(),
```

Comment:
We should have the same signature for `get_dummy_raw_counterparty()` as we have for channel.

Reply:
Can you expand on this suggestion? `get_dummy_raw_channel_end` has similar signature with `get_dummy_raw_counterparty`.

Reply:
looks good now, i was referring to channel vs connection.